devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy

```diff
@@ -79,6 +83,8 @@ def parse(String description) {
                 map = translateZoneStatus(zs)
             } else if (descMap?.clusterInt == OCCUPANCY_SENSING_CLUSTER && descMap.attrInt == OCCUPANCY_SENSING_OCCUPANCY_ATTRIBUTE && descMap?.value) {
                 map = getMotionResult(descMap.value == "01" ? "active" : "inactive")
+            } else if (descMap?.clusterInt == ANALOG_INPUT_BASIC_CLUSTER && descMap.attrInt == ANALOG_INPUT_BASIC_PRESENT_VALUE_ATTRIBUTE && descMap?.value) {
+                map = getAnalogInputResult(Long.parseLong(descMap.value, 16))                
```

Comment:
What's the example input here ? (descMap.value)
Did You try `Integer.parseInt(descMap.value,16)` ?
That way maybe You wouldn't have to parse it to long, convert to float and round this value in `getAnalogInputResult`